### PR TITLE
Update state recovery initialization command

### LINKED ISSUE #7768: Update state recovery initialization command
### Description

This command is used to initialize a terminated state account or update the state root stored in it if it was not set yet. It can only be submitted on a sidechain.

- Update params [schema](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#parameters-3)
- Update [verification](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#verification-3)
```python
def verify(trs: Transaction) -> None:
    if trs.params.chainID == CHAIN_ID_MAINCHAIN or trs.params.chainID == ownChainAccount.chainID:
        raise Exception("Chain ID is not valid.")

    # The commands fails if the sidechain is already terminated on this chain.
    if terminatedStateAccount(trs.params.chainID) exists and terminatedStateAccount(trs.params.chainID).initialized == True:
        raise Exception("Sidechain is already terminated.")

    sidechainAccount = decode(chainAccountSchema, trs.params.sidechainAccount)
    # The commands fails if the sidechain is not terminated and did not violate the liveness requirement.
    if sidechainAccount.status != CHAIN_STATUS_TERMINATED
        and chainAccount(CHAIN_ID_MAINCHAIN).lastCertificate.timestamp - sidechainAccount.lastCertificate.timestamp <= LIVENESS_LIMIT:
        raise Exception("Sidechain is not terminated.")

    queryKey = STORE_PREFIX_INTEROPERABILITY + SUBSTORE_PREFIX_CHAIN_DATA + sha256(trs.params.chainID)

    query = {
        "key": queryKey,
        "value": sha256(trs.params.sidechainAccount),
        "bitmap": trs.params.bitmap
    }

    proofOfInclusion = { "siblingHashes": trs.params.siblingHashes, "queries" : [query] }

    if terminatedStateAccount(trs.params.chainID) exists:
        if SMTVerify([queryKey], proofOfInclusion, terminatedStateAccount(trs.params.chainID).mainchainStateRoot) == False:
            raise Exception("State recovery initialization proof of inclusion is not valid.")
    else:
        if SMTVerify([queryKey], proofOfInclusion, chainAccount(CHAIN_ID_MAINCHAIN).lastCertificate.stateRoot) == False:
            raise Exception("State recovery initialization proof of inclusion is not valid.")
```
- Update [execution](https://github.com/LiskHQ/lips/blob/main/proposals/lip-0054.md#execution-3)
```python
def execute(trs: Transaction) -> None:
    sidechainAccount = decode(chainAccountSchema, trs.params.sidechainAccount)
    if terminatedStateAccount(trs.params.chainID) exists:
        terminatedStateAccount(trs.params.chainID).stateRoot = sidechainAccount.lastCertificate.stateRoot
        terminatedStateAccount(trs.params.chainID).mainchainStateRoot = EMPTY_BYTES
        terminatedStateAccount(trs.params.chainID).initialized = True
    else:
        createTerminatedStateAccount(trs.params.chainID, sidechainAccount.lastCertificate.stateRoot)
```

### Acceptance Criteria

- Should have all the unit tests 

### Additional Information

- Related LIP PR - https://github.com/LiskHQ/lips/pull/172/

### PR DESCRIPTION
**Pull request by ishantiw**
### What was the problem?

This PR resolves #7768

### How was it solved?

- Update schema
- Update verify for error messages and smt verify update
- Update tests

### How was it tested?

`npm run test:unit state_recovery_init.spec.ts`

### CODE REVIEW

**Review by shuse2 — APPROVED** [review]

File: framework/src/modules/interoperability/types.ts [hunk 1]
```diff
@@ -323,7 +323,7 @@ export interface StateRecoveryParams {
 
 export interface StateRecoveryInitParams {
 	chainID: Buffer;
-	sidechainChainAccount: Buffer;
+	sidechainAccount: Buffer;
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/types.ts:326]: this not part of LIP so we don't have to

File: framework/src/modules/interoperability/sidechain/commands/state_recovery_init.ts [hunk 2]
```diff
@@ -107,11 +107,11 @@ export class StateRecoveryInitializationCommand extends BaseInteroperabilityComm
 				};
 			}
 		} else {
-			const verified = sparseMerkleTree.verify(
-				[interopAccKey],
-				proofOfInclusion,
+			const smt = new SparseMerkleTree();
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/sidechain/commands/state_recovery_init.ts:110]: i think we can keep the smt.verify each time

File: framework/src/modules/interoperability/sidechain/commands/state_recovery_init.ts [hunk 3]
```diff
@@ -94,11 +94,11 @@ export class StateRecoveryInitializationCommand extends BaseInteroperabilityComm
 			if (!terminatedStateAccount.mainchainStateRoot) {
 				throw new Error('Sidechain account has missing property: mainchain state root');
 			}
-			const verified = sparseMerkleTree.verify(
-				[interopAccKey],
+			const smt = new SparseMerkleTree();
+			const verified = await smt.verify(
+				terminatedStateAccount.stateRoot,
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/sidechain/commands/state_recovery_init.ts:99]: good catch

File: framework/src/modules/interoperability/sidechain/commands/state_recovery_init.ts [hunk 4]
```diff
@@ -130,7 +130,7 @@ export class StateRecoveryInitializationCommand extends BaseInteroperabilityComm
 		const { params } = context;
 		const sidechainChainAccount = codec.decode<ChainAccount>(
 			chainAccountSchema,
-			params.sidechainChainAccount,
+			params.sidechainAccount,
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/src/modules/interoperability/sidechain/commands/state_recovery_init.ts:133]: For point 3, i think we try to follow early return format, where if the condition is met we return and it is unnecessary to have else block

File: framework/test/unit/modules/interoperability/sidechain/commands/state_recovery_init.spec.ts [hunk 5]
```diff
@@ -96,10 +97,10 @@ describe('Mainchain StateRecoveryInitializationCommand', () => {
 			chainID: utils.intToBuffer(3, 4),
 			bitmap: Buffer.alloc(0),
 			siblingHashes: [],
-			sidechainChainAccount: sidechainChainAccountEncoded,
+			sidechainAccount: sidechainChainAccountEncoded,
```
[the remark this answers is not in the document]
Comment (ishantiw) [reply framework/test/unit/modules/interoperability/sidechain/commands/state_recovery_init.spec.ts:100]: we can resolve it in another issue that will touch this code base

**Pull request merged**